### ISSUE TITLE
Buffer creation interface

### ISSUE THREAD

**Issue by bramtayl** [JuliaAudio/SampledSignals.jl#83]
Currently, if you want to write from a SampledSignal to PortAudio, you will end up maintaining two buffers. SampledSignals will make a buffer, PortAudio will make a buffer, and all data will be copied back and forth between these buffers. It seems possible for SampledSignals to just use the PortAudioBuffer. One way of doing this would be to have some sort of public interface, maybe a `make_buffer` function, that defaults to allocating a regular array. Then, PortAudio could overload that function and say "you can share my buffer".

**Comment by rob-luke** [comment]
> It seems possible for SampledSignals to just use the PortAudioBuffer

Are you implying that SampledSignals should have PortAudio as a dependency? SampledSignals has broad applications and is used beyond just audio, for example for neuroimaging signals. So I don't think it's appropriate to add PortAudio as a requirement.

Or are you suggesting some new API or functionality for SampledSignals?

**Comment by bramtayl** [comment]
> Or are you suggesting some new API or functionality for SampledSignals?

Yes, this one. Sorry if it wasn't clear.

**Comment by rob-luke** [comment]
> It seems possible for SampledSignals to just use the PortAudioBuffer

Yeah I think I was confused by this sentence. I though you were implying that SampledSignals should use PortAudio output, which of course it cant, I dont want my EEG code having PortAudio as a dependency.

Ok so what exactly are you proposing as the change to SampledSignals? I think just keep in mind that the two projects are relatively independent, so it should have a general API.

**Comment by bramtayl** [comment]
Maybe something like a `make_buffer` function that defaults to this line: https://github.com/JuliaAudio/SampledSignals.jl/blob/0d9954409d877f2bd74671d31fd1cdd7acc548c1/src/SampleStream.jl#L136 and that sinks can overload?
